src/agent/onefuzz-supervisor/src/buffer.rs

```diff
@@ -54,22 +54,23 @@ mod tests {
 
         assert!(buf.data().is_empty());
 
-        buf.write(&[1, 2, 3]).unwrap();
+        buf.write_all(&[1, 2, 3]).unwrap();
```

Comment:
Our impl of `write()` is actually the method under test, so I think it is most appropriate to suppress the linter here. `Write::write_all()` ends up being equivalent, but it _would_ mask a change.

Reply:
If our implementation of write is under test, then should we test the result?

Reply:
I think so, so we can be sure it plays nice with wrappers. So for example, I'd suggest that this line becomes:
```suggestion
        assert_eq!(buf.write(&[1, 2, 3]), Ok(3));
```